defi/src/utils/craftParentProtocol.ts

```diff
@@ -102,6 +100,27 @@ export default async function craftParentProtocol({
     };
   }
 
+  return craftParentProtocolInternal({ parentProtocol, skipAggregatedTvl, isHourlyTvl, childProtocolsTvls, });
+}
+
+export async function craftParentProtocolInternal({
+  parentProtocol,
+  skipAggregatedTvl,
+  childProtocolsTvls,
+  isHourlyTvl,
+  fetchMcap,
+}: {
+  parentProtocol: IParentProtocol;
+  skipAggregatedTvl: boolean;
+  isHourlyTvl: Function;
+  fetchMcap?: Function;
```

Comment:
if this is undefined wont line 416 fail?

Reply:
I am setting it in line 120 if empty: https://github.com/DefiLlama/defillama-server/blob/api-v2/defi/src/utils/craftParentProtocol.ts#L120

Reply:
> if this is undefined wont line 416 fail?

I am setting it in line 120 if empty: 
https://github.com/DefiLlama/defillama-server/blob/api-v2/defi/src/utils/craftParentProtocol.ts#L120

Reply:
im stupid